dwave/optimization/_model.pyx

```diff
@@ -619,19 +641,46 @@ cdef class _Graph:
         return self._graph.remove_unused_nodes()
 
     def state_size(self):
-        """An estimate of the size, in bytes, of all states in the model.
+        r"""Return an estimate of the size, in bytes, of a model state.
+
+        For a model encoding several array operations, the state of each array
+        must be held in memory. This method returns an estimate of the total
+        memory needed to hold a state for every symbol in the model.
 
-        Iterates over the model's states and totals the sizes of all.
+        The number of bytes returned by this method is only an estimate. Some
+        symbols hold additional information that is not accounted for.
 
         Examples:
-            This example estimates the size of a model's states.
+            This example estimates the size of a model state.
+            In this example a single value is added to a :math:`5\times4` array.
+            The output of the addition is also a :math:`5\times4` array.
+            Each element of each array requires :math:`8` bytes to represent
+            in memory.
+            Therefore the total state size is :math:`(5*4 + 1 + 5*4) * 8 = 328`
+            bytes.
 
             >>> from dwave.optimization.model import Model
             >>> model = Model()
-            >>> c = model.constant([1, 5, 8.4])
-            >>> i = model.integer(20, upper_bound=100)
-            >>> model.state_size()
-            184
+            >>> i = model.integer((5, 4))  # 5x4 array of integers
+            >>> c = model.constant(1)      # one scalar value
+            >>> y = i + c                  # 5x4 array of values
+            >>> model.state_size()         # (5*4 + 1 + 5*4) * 8 bytes
+            328
+
+        See also:
+            :meth:`Symbol.state_size()` An estimate of the size of a symbol's
+            state.
+
+            :meth:`ArraySymbol.state_size()` An estimate of the size of an array
+            symbol's state.
+
+            :meth:`Model.decision_state_size()` An estimate of the size of a
+            model's decision states.
+
+            :ref:`properties_solver_properties` The properties of the
```

Comment:
Minor (ignore if you wish): Without a colon after the "[Solver Properties](https://docs.dwavesys.com/docs/latest/c_solver_properties.html)" this reads a bit strange. You can change to something like 

```
The  :ref:`properties  documentation \<properties_solver_properties\>` for the
```

(where the backslashes are due to github rendering) or add a colon

Reply:
For this one I am incline to ignore. It keeps the pattern of the previous ones and IMO is clear enough in context.